api/queue/work/model_service_deployment.go

```diff
@@ -215,7 +215,7 @@ func (depl *ModelServiceDeployment) generateModelOptions(ctx context.Context, mo
 	modelOpt := &models.ModelOption{}
 	switch model.Type {
 	case models.ModelTypePyFunc, models.ModelTypePyFuncV3:
-		imageRef, err := depl.ImageBuilder.BuildImage(ctx, model.Project, model, version, endpoint.ImageBuilderResourceRequest)
+		imageRef, err := depl.ImageBuilder.BuildImage(ctx, model.Project, model, version, endpoint.ImageBuilderResourceRequest, nil)
```

Comment:
whats the motivation behind adding backofflimit when all the caller are going to pass a nil?

Reply:
Passing nil there for backward compatibility, so if they pass nil, Merlin will use the default value from the configuration (https://github.com/caraml-dev/merlin/blob/main/api/config/config.go#L225).

If SDK specifies a number, it will get propagated to the API and kaniko job spec.

For more background, the original reason we added this retry for the image building job is to retry when the pip installation failed due to an issue that can be recovered (e.g. network error).